### ISSUE TITLE
thermal runaway false positve trigger

### ISSUE THREAD

**Issue by Sungod3000** [MarlinFirmware/Marlin#3302]
Hi,

take this with a grain of salt because I had the same issue in the last marlin (RC1) so I disabled it but now the default settings are triggering print stop ca 1 min into printing. I extended the parameters to 90secs and 10C but still got the same shutdown time.

**Comment by thinkyhead** [comment]
If you're testing with `RCBugFix` the defaults to check fast-enough-heating have been relaxed a little. It used to be 4 degrees in 16 seconds. Now it's more like 2 degrees in 20 seconds. And we've patched some bugs around the wait-for-heating routines too. Overall it should be better now, so give it a try and see if the new defaults are good enough on your machine.

**Comment by maxwondercorn** [comment]
The latest `RCBugFix` commit fixed these issues for me.

## Update:

I spoke too soon.  The issues during heat up are fixed and I can now do some prints that I couldn't do before, but as soon as I start putting out a large volume of filament the temperate can't keep up.  The first layer lays out well but when I start the second layer which runs at a higher speed the PID loop looses control.  You can see it tries to recover but it goes into thermal runaway before it can recover. (See the first screenshot)

I've run the same experiment with thermal protection disabled and it drops, doesn't recover and goes into a temperature control mode below the setpoint. (See other screen shots)  If I put a DVM across the heater terminals it's obvious the oscillations are being controlled.

The firmware never tries to drive the temperature fully to the setpoint.  A step change in the set point causes the oscillations to shift but again the system never tries to reach the setpoint.

While I originally thought this was a loose wire - both for the thermister and the catridge heater, I've gone through that excircise to prove that it's not a wiring issue. The Ramps board has been replaced, but I don't have a spare Mega2600.  I've tuned the hot end and it doesn't make a difference - when it's on setpoiint the PID holds a steady temperature.

I haven't tried this with thermal protection disabled

![screen shot 2016-04-04 at 12 36 29 am](https://cloud.githubusercontent.com/assets/18253733/14238333/ea475f4e-f9fd-11e5-9b57-6b839de0f29e.png)

![screen shot 2016-04-04 at 1 13 42 am](https://cloud.githubusercontent.com/assets/18253733/14239901/5fe2d778-fa0e-11e5-842f-faca25fef084.png)

![screen shot 2016-04-04 at 1 20 44 am](https://cloud.githubusercontent.com/assets/18253733/14239923/883dbe04-fa0e-11e5-9a67-6dc8d5efa09c.png)

**Comment by jbrazio** [comment]
@maxwondercorn may I suggest to set `PID_FUNCTIONAL_RANGE` to `250` and repeat the tests ? The default value of `10` means that the PID is only active when the temp is `±10C` from the setpoint.

**Comment by maxwondercorn** [comment]
@jbrazio Okay, I was literally up all nite working on this and figured it out.  Definitely a hard and probably a stupid lesson... 

I rolled back to RC2 and the problem was still occurring even though it wasn't a problem before. While playing around I sliced a part in a different slicer (Slic3r) and the problem went away.    Tried again with normal slicer - Matter Control - problem came back. More mystery - how can a slicer make a difference?

As I was physically looking over the printer for the hundredth time I noticed my parts fan had gotten knocked aside and was pointing directly at the heater block and it was RUNNING! 

I have a very large printer and have to go behind it to even see if the part fan is running.   Since it only really runs when doing bridges I just let it point wherever until I know it's going to run.  Matter Control (normal slicer) had an update a week or so back and now it was running the fan starting on the second layer where it hadn't been before!  Same parts, different behaviour.

I've been looking at this problem for a while and can't believe I missed this.  The good part is I did learn a lot more about the inner workings of Marlin, so I guess that'll make me a better troubleshooter...

thanks.
